### PR TITLE
[uss_qualifier] Update USSQualifierConfiguration

### PR DESCRIPTION
**Pull request by BenjaminPelletier**
Currently, specifications to uss_qualifier to generate artifacts are provided via command line flags.  However, the values reasonable for these flags depend heavily on the specified test configuration -- it wouldn't make sense, for instance, to generate tested requirements for a NetRID Service Provider role when only an F3548-21 test was conducted.  This PR resolves this issue by establishing a new `USSQualifierConfigurationV1` which contains both a `TestConfiguration` (as previously defined) and a new `ArtifactsConfiguration`.  This way, artifact generation can be defined along side the test configuration that generates the report that will be used in that artifact generation.

To make configuration format changes more backwards-compatible in the future, versioning is introduced by placing `USSQualifierConfigurationV1` within a new master `USSQualifierConfiguration` which has a single `v1` field currently.  A new version would simply add another `v2` field.

The configuration is also expanded to support any test action, not just a test suite.  This will make testing a single scenario easier than it currently is, as doing so will no longer require creating a suite pointing just to that scenario.

This PR makes a breaking change to configuration formats, changing the expected schema from `TestConfiguration` to `USSQualifierConfiguration`.  All configurations tracked in the repository have been tested, including:
* configurations.dev.local_test
* configurations.dev.f3548
* configurations.dev.generate_rid_test_data
* configurations.dev.self_contained_f3548
* configurations.dev.faa.uft.local_message_signing
* configurations.dev.non_docker.local_test (for validity, not for success)

Note that the diffs will look smaller if, on the "Files changed" tab for the PR, the gear icon is clicked to check "Hide whitespace".

### CODE REVIEW

File: monitoring/uss_qualifier/main.py [hunk 1]
```diff
@@ -27,67 +31,67 @@ def parseArgs() -> argparse.Namespace:
...
+    if config.test_run:
+        report = execute_test_run(config.test_run)
+        if config.artifacts and config.artifacts.report_path:
+            print(f"Writing report to {config.artifacts.report_path}")
+            with open(config.artifacts.report_path, "w") as f:
                 json.dump(report, f, indent=2)
-    elif args.report is not None:
-        with open(args.report, "r") as f:
+    elif config.artifacts and config.artifacts.report_path:
+        with open(config.report_path, "r") as f:
```
[the remark this answers is not in the document]
Comment (BenjaminPelletier) [reply monitoring/uss_qualifier/main.py:73]: Yes, definitely -- thanks.

**Comment by BenjaminPelletier** [comment]
Yes, I think versioning makes sense.  There are a number of ways to do it with various degrees of inconvenience -- I'm hoping that, after this change, we shouldn't need to restructure things much.  When fields are just added, backwards compatibility is easy.  I think the most robust approach would probably be to have a top-level object that has a single `v1` field pointing to `USSQualifierConfigurationV1` (which would be identical to `USSQualifierConfiguration` currently).  Then, a new schema would simply populate a `v2` field and so on.

It's not great to keep changing the schema ad-hoc like this PR does, so given that's a good thing to do, it seems like it probably makes sense to do it in this PR.  I'll make that addition and then request a second look.

**Comment by BenjaminPelletier** [comment]
@barroco , I've added versioning to configuration files in this PR and so would appreciate a second look with those non-trivial changes (though they are a small number of lines of code; see commits).  #919 conflicts with these changes (it has a new configuration file in the format prior to this PR), so I'm happy to have #919 merged first and then I can update the configuration format in this PR after #919 is merged.

**Comment by BenjaminPelletier** [comment]
I've further expanded the scope of this PR in an additional commit allowing targeting of a test scenario directly in the configuration without needing to go through an intermediate test suite.  The PR title and description have been updated accordingly.  @barroco , please treat these changes as a new PR based on the previous one (though see commits for what has changed versus what hasn't)

File: monitoring/mock_uss/run_locally_geoawareness.sh [hunk 2]
```diff
@@ -21,7 +21,7 @@ if [ "$TEST" == "true" ]; then
 fi
 
 # shellcheck disable=SC2086
-docker run ${docker_args} --name mock_uss_scdsc \
+docker run ${docker_args} --name mock_uss_geoawareness \
```
Comment (barroco) [inline monitoring/mock_uss/run_locally_geoawareness.sh:24]: Thank you for this 👍

**Review by barroco — APPROVED** [review]

**Pull request merged**